### PR TITLE
Routing

### PR DESCRIPTION
**Pull request by alexwaeseperlman**
Sorry I took so long. This PR removes all the flex layout logic and stores enough state to create shareable links the url. Currently the urls are pretty long since they include all the data about loaded genes. For example:
```
http://localhost:5173/?gene=%22AT1G01010%22&view=%22publication-viewer%22&genes=%5B%7B%22id%22%3A%22AT1G01010%22%2C%22species%22%3A%22Arabidopsis%22%2C%22annotation%22%3A%22NAC+domain+containing+protein+1%22%2C%22aliases%22%3A%5B%22ANAC001%22%2C%22NAC001%22%2C%22NTL10%22%5D%7D%5D
```

I don't think this is an issue, since we can include a url shortener for cases where people want to share things. Another option would be to include a list of gene ids instead of the data itself, and download the rest of the data on page load.

### CODE REVIEW

File: Eplant/UI/LeftNav/index.tsx [hunk 1]
```diff
@@ -29,8 +28,7 @@ export function LeftNav(props: {
   const [geneticElements, setGeneticElements] = useGeneticElements()
   const [darkMode, setDarkMode] = useDarkMode()
 
-  const panesDispatch = usePanesDispatch()
-  const activeID = useActiveId()[0]
+  const setActiveGeneId = useSetActiveGeneId()
 
   React.useEffect(() => {
```
Comment (zkdan) [inline Eplant/UI/LeftNav/index.tsx]: Can you take out the `React.`s wherever you find them, please? If you get an error, the eslint config file should have a rule (I think `'react/react-in-jsx-scope': 'off',`) that'll fix it.

File: Eplant/UI/Layout/ViewContainer/index.tsx [hunk 2]
```diff
@@ -54,15 +54,13 @@ export function ViewContainer<T, S, A>({
 
   const [viewingCitations, setViewingCitations] = React.useState(false)
```
Comment (zkdan) [inline Eplant/UI/Layout/ViewContainer/index.tsx]: remove `React.`

File: Eplant/config.ts [hunk 3]
```diff
@@ -6,16 +6,20 @@ export type EplantConfig = {
...
   userViews: [],
   views: [],
-  tabHeight: 48,
+
   rootPath: '',
+  defaultSpecies: '',
+  defaultView: '',
 })
 
 export const useConfig = () => React.useContext(Config)
```
Comment (zkdan) [inline Eplant/config.ts:33]: remove `React.`

**Review by zkdan — DISMISSED** [review]
At the meeting today we talked to Nick who said he would prefer a shorter URL and using the URL params to load the information (as opposed to having it all in the URL and shortening it).

I don't want to block the work you did taking the panes out, so my suggestion is to roll this PR back a couple commits to only include the removal of the panes so we can merge it in. Then routing can be a separate PR by you or someone else if you don't have the time/inclination.

File: Eplant/Eplant.tsx [hunk 4]
```diff
@@ -1,40 +1,76 @@
...
+  useEffect(() => {
+    if (loaded) {
+      updateColors(theme)
+    }
+  }, [theme, loaded])
   return (
     <>
       <Sidebar />
-      <EplantLayout />
+      <Box
```
Comment (Yukthiw) [inline Eplant/Eplant.tsx:31]: Is there a reason we aren't breaking this out into a separate component?

File: Eplant/UI/Layout/ViewContainer/index.tsx [hunk 5]
```diff
@@ -36,27 +37,26 @@ import ViewOptions from './ViewOptions'
...
-export function ViewContainer<T, S, A>({
+export function ViewContainer<T = any, S = any, A = any>({
   view,
-  setView,
   gene,
   ...props
 }: {
   view: View<T, S, A>
-  setView: (view: View) => void
+  setView?: (view: View) => void
```
Comment (Yukthiw) [inline Eplant/UI/Layout/ViewContainer/index.tsx:46]: Nit: Why is this optional?

**Review by Yukthiw — APPROVED** [review]
LGTM

**Pull request merged**